jpa/src/main/java/org/jboss/as/jpa/processor/PersistenceUnitServiceHandler.java

```diff
@@ -171,12 +171,14 @@ private static void handleWarDeployment(DeploymentPhaseContext phaseContext, boo
             PersistenceUnitMetadataHolder holder;
             ArrayList<PersistenceUnitMetadataHolder> puList = new ArrayList<PersistenceUnitMetadataHolder>(1);
 
+            String deploymentRootName = null;
             // handle persistence.xml definition in the root of the war
             if (deploymentRoot != null &&
                 (holder = deploymentRoot.getAttachment(PersistenceUnitMetadataHolder.PERSISTENCE_UNITS)) != null &&
                 holder.getPersistenceUnits().size() > 0) {
                 // assemble and install the PU service
                 puList.add(holder);
+                deploymentRootName = deploymentRoot.getRootName()
```

Comment:
looks like a missing semicolon here is causing the next build error.  I didn't try to build your change either.